src/gameengine.js

```diff
@@ -631,33 +664,37 @@ GameEngine.prototype.start = function () {
-        if(this.agents[i].entity.controllable === true) {
-            if(this.pressRight && !this.pressLeft) this.agents[i].readInput("right");
-            if(this.pressDown) this.agents[i].readInput("down");
-            if(this.pressUp) this.agents[i].readInput("up");
-            if(this.pressLeft && !this.pressRight) this.agents[i].readInput("left");
-            if(this.pressLeft && this.pressRight) this.agents[i].readInput("none");
-            if(this.pressN) this.agents[i].readInput('n');
-            if(this.pressSpace) this.agents[i].readInput("space");
-
-            if(this.pressRight && this.pressLeft) {
-                this.agents[i].readInput("left_released");
-                this.agents[i].readInput("right_released");
-            }
-            if(!this.pressUp) this.agents[i].readInput("up_released");
-            if(!this.pressLeft) this.agents[i].readInput("left_released");
-            if(!this.pressRight) this.agents[i].readInput("right_released");
-            if(!this.pressSpace) this.agents[i].readInput("space_released");
-            if(!this.pressRight && !this.pressLeft) this.agents[i].readInput("left_and_right_released");
-
-            if(!this.pressLeft && !this.pressRight && !this.pressDown && !this.pressUp && !this.pressSpace) this.agents[i].readInput("none");
-        }
+    if (this.visible && this.hasFocus && !this.paused) {
+      for(var i = 0; i < this.agents.length; i++) {
+          if(this.agents[i].entity.controllable === true) {
+              if(this.pressRight && !this.pressLeft) this.agents[i].readInput("right");
+              if(this.pressDown) this.agents[i].readInput("down");
+              if(this.pressUp) this.agents[i].readInput("up");
+              if(this.pressLeft && !this.pressRight) this.agents[i].readInput("left");
+              if(this.pressLeft && this.pressRight) this.agents[i].readInput("none");
+              if(this.pressN) this.agents[i].readInput('n');
+              if(this.pressSpace) this.agents[i].readInput("space");
+
+              if(this.pressRight && this.pressLeft) {
+                  this.agents[i].readInput("left_released");
+                  this.agents[i].readInput("right_released");
+              }
+              if(!this.pressUp) this.agents[i].readInput("up_released");
+              if(!this.pressLeft) this.agents[i].readInput("left_released");
+              if(!this.pressRight) this.agents[i].readInput("right_released");
+              if(!this.pressSpace) this.agents[i].readInput("space_released");
+              if(!this.pressRight && !this.pressLeft) this.agents[i].readInput("left_and_right_released");
+
+              if(!this.pressLeft && !this.pressRight && !this.pressDown && !this.pressUp && !this.pressSpace) this.agents[i].readInput("none");
+          }
+      }
+
+      this.clockTick = this.timer.tick();
+      this.update();
+      this.draw();
+    } else {
+        // Deal with being paused here
```

Comment:
Correct. I think the original idea was that it would be a splash screen, but I don't mind the player looking at the game and thinking while it's paused.